CampusTour/Model/LocationProvider.swift

```diff
@@ -14,8 +14,15 @@ enum LocationError: Error {
 }
 
 class LocationProvider: NSObject {
+    typealias LocationListener = (CLLocation) -> ()
+    struct LocationListenerWrapper {
+        let listener: LocationListener
+        let repeats: Bool
+    }
+    
     private var _currentLocation: CLLocation?
     private let locationManager: CLLocationManager
+    private var locationListeners: [LocationListenerWrapper] = []
```

Comment:
Not super familiar with locationListeners--why do we need a list here instead of a single reference?